### ISSUE TITLE
Enhancement, Whitelistning

### ISSUE THREAD

**Issue by mljdivemaster** [guilhemmarchand/trackme#27]
Is it possible to have a function as white listing? everything is tracked by default but some sources you can mark in a white list to keep track on what should be monitored?

**Comment by guilhemmarchand** [comment]
Hi,

Good point, yes I can work on a whitelisting feature similar to the blacklisting methodology, driven via the UI.

**There are actually macros definition that are being called at the root of the searches:**

```
# used as the top of the populating searches
[trackme_tstats_main_filter]
definition = sourcetype!="stash" sourcetype!="*too_small"
iseval = 0

```

**Extracted from the searches:**

```
| tstats max(_indextime) as current_data_last_ingest, max(_time) as current_data_last_time_seen where index=* sourcetype=* `trackme_tstats_main_filter
```

For the time being, you could customise the macro have a constraint that whitelist indexes, source, sourcetypes or hosts that shall only be included.

Therefore I will work a OOTB feature.

**Comment by mljdivemaster** [comment]
Perfect!

I think for a big installation with a lot of data sources, there is a need to keep track of everything but some changes are more important than other. If the Active Directory logs are missing for an example, that should trigger an alert. If a end user asks where their data is, we can show how it looks like historically. It will help you to keep track of what is normal and what is missing

**Comment by guilhemmarchand** [comment]
Hi @mljdivemaster !

It's been a bit more time than expected to implement the whitelisting, but finally it is available in the very last version 1.0.32.

Whitelisting is available for data sources and hosts, driven via the UI:

https://trackme.readthedocs.io/en/latest/configuration.html#whitelisting-and-blacklisting

Please feel free to let me know if you have anymore question or requests.

Guilhem
